js/app/test/gallery_component_events.spec.ts

```diff
@@ -18,6 +18,7 @@ test("Gallery preview mode displays all images correctly.", async ({
 test("Gallery select event returns the right value", async ({ page }) => {
 	await page.getByRole("button", { name: "Run" }).click();
 	await page.getByLabel("Thumbnail 2 of 3").click();
+	await page.waitForTimeout(200);
 	expect(await page.getByLabel("Select Data")).toHaveValue(
 		"https://gradio-builds.s3.amazonaws.com/assets/lite-logo.png"
 	);
```

Comment:
Could you try removing the timeout and doing this instead. See if the results are any different:

```suggestion
	await expect(page.getByLabel("Select Data")).toHaveValue(
		"https://gradio-builds.s3.amazonaws.com/assets/lite-logo.png"
	);
```

Reply:
Actually I've just checked the other tests and I'm almost certain this is the issue.

The way retries in Playwright work is that you pass in a `Locator` (with is what this `page.getByLabel(...)` call returns) and it can use that structure to requery if the value isn't ready yet. In order for those retries to take affect, we need to await the expectation itself rather than the locator.

Assuming this works, could you update the other tests in this file to match?

Reply:
That wasn't very clear I'll try again.

The `Locator` is a reference to some part of the dom. Its the `toHaveValue(...)` that retries. In order for it to not immediately return we need to await it, to give a chance to actually retry.

Reply:
Ohhh ok. I'm pretty sure you have explained this to me before. Will fix. Thanks so much!

Reply:
Looking at the other test files, it seems the approach of "await everything just incase" seems to work well enough because there are definitely some unnecessary awaits in those but they don't seem too fléké.

Although i do sometime have issue with the streaming chatbot test locally (never in ci).